tests/CMakeLists.txt

```diff
@@ -16,20 +16,21 @@ if (CODE_COVERAGE)
     evc_include(CodeCoverage)
 endif()
 
-add_executable(test_state_allocator state_allocator.cpp)
-target_link_libraries(test_state_allocator
+set(TEST_TARGET_NAME ${PROJECT_NAME}_tests)
```

Comment:
`TEST_TARGET_NAME` is to general here, especially when having multiple tests.  There should be a `TEST_PREFIX` or something similar, which gets prepended to `state_allocator`

Reply:
In other repos we re-use that TEST_TARGET_NAME variable for multiple test cases, so setting it to something like
```cmake
set(TEST_TARGET_NAME ${PROJECT_NAME}_state_allocator_tests)
```

and later
```cmake
set(TEST_TARGET_NAME ${PROJECT_NAME}_some_other_target_name_tests)
```

If we just use a `TEST_PREFIX` we still have to repeat the appended test name, but an argument could be made for unique variable names